demo/ARContainerViewController.swift

```diff
@@ -92,9 +89,29 @@ final class ARContainerViewController: UIViewController {
 }
 
 extension ARContainerViewController: NavigationServiceDelegate {
-    
+
     func navigationService(_ service: NavigationService, didUpdate progress: RouteProgress, with location: CLLocation, rawLocation: CLLocation) {
         instructionsLabel.isHidden = false
         instructionsLabel.text = progress.currentLegProgress.currentStep.instructions
+
+        if routeHasChanged {
+            routeHasChanged = false
+            navigationDelegate?.navigationService(didUpdate: Route(route: progress.route))
+        }
+
+        if progress.currentLegProgress.userHasArrivedAtWaypoint {
+            routeHasChanged = true
```

Comment:
Why we want situation `navigationDelegate?.navigationService(didUpdate: Route(route: progress.route))` will be called above when `userHasArrivedAtWaypoint`?

Reply:
removed